File: javaproject/app/src/main/java/com/marcosvbras/robomarket/selectrobot/ui/SelectRobotActivity.java
package com.marcosvbras.robomarket.selectrobot.ui;

import android.arch.lifecycle.ViewModelProviders;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.widget.SearchView;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.marcosvbras.robomarket.R;
import com.marcosvbras.robomarket.app.BaseActivity;
import com.marcosvbras.robomarket.databinding.ActivitySelectRobotBinding;
import com.marcosvbras.robomarket.selectrobot.viewmodel.SelectRobotViewModel;
import com.marcosvbras.robomarket.selectrobot.viewmodel.SelectRobotViewModelFactory;

public class SelectRobotActivity extends BaseActivity {

    private ActivitySelectRobotBinding activityBinding;
    private SearchView searchView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activityBinding = DataBindingUtil.setContentView(this, R.layout.activity_select_robot);
        activityBinding.setViewModel(createViewModel());
        activityBinding.executePendingBindings();
        setToolbar(R.id.top_toolbar, true);
    }

    private SelectRobotViewModel createViewModel() {
        return ViewModelProviders.of(this, new SelectRobotViewModelFactory(this))
                .get(SelectRobotViewModel.class);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_home, menu);
        menu.findItem(R.id.menu_add).setVisible(false);
        menu.findItem(R.id.menu_logout).setVisible(false);
        MenuItem menuItemSearch = menu.findItem(R.id.menu_search);
        searchView = (SearchView)menuItemSearch.getActionView();
        searchView.setOnQueryTextListener(onSearchListener());
        searchView.setOnCloseListener(() -> {
            activityBinding.getViewModel().listRobots(null);
            return false;
        });

        return super.onCreateOptionsMenu(menu);
    }

    private SearchView.OnQueryTextListener onSearchListener() {
        return new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                activityBinding.getViewModel().listRobots(newText);
                return true;
            }
        };
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == android.R.id.home)
            finish();

        return super.onOptionsItemSelected(item);
    }
}

File: project/app/src/main/java/com/marcosvbras/robomarket/flows/selectrobot/ui/SelectRobotActivity.kt
package com.marcosvbras.robomarket.flows.selectrobot.ui

import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.widget.SearchView
import android.view.Menu
import android.view.MenuItem
import com.marcosvbras.robomarket.R
import com.marcosvbras.robomarket.app.BaseActivity
import com.marcosvbras.robomarket.databinding.ActivitySelectRobotBinding
import com.marcosvbras.robomarket.flows.selectrobot.viewmodel.SelectRobotViewModel
import com.marcosvbras.robomarket.flows.selectrobot.viewmodel.SelectRobotViewModelFactory

class SelectRobotActivity : BaseActivity() {

    private var activityBinding: ActivitySelectRobotBinding? = null
    private var searchView: SearchView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityBinding = DataBindingUtil.setContentView(this, R.layout.activity_select_robot)
        setToolbar(R.id.top_toolbar, true)
        activityBinding?.viewModel = createViewModel()
        activityBinding?.executePendingBindings()
    }

    private fun createViewModel(): SelectRobotViewModel {
        return ViewModelProviders.of(this, SelectRobotViewModelFactory(this))
                .get(SelectRobotViewModel::class.java)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_home, menu)
        menu.findItem(R.id.menu_add).isVisible = false
        menu.findItem(R.id.menu_logout).isVisible = false
        val menuItemSearch = menu.findItem(R.id.menu_search)
        searchView = menuItemSearch.actionView as SearchView
        searchView?.setOnQueryTextListener(onSearchListener())
        searchView?.setOnCloseListener {
            activityBinding?.viewModel?.listRobots(null)
            false
        }

        return super.onCreateOptionsMenu(menu)
    }

    private fun onSearchListener(): SearchView.OnQueryTextListener {
        return object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                return false
            }

            override fun onQueryTextChange(newText: String): Boolean {
                activityBinding?.viewModel?.listRobots(newText)
                return true
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home)
            finish()

        return super.onOptionsItemSelected(item)
    }
}